strongbox-storage/strongbox-storage-layout-providers/strongbox-storage-maven-layout/strongbox-storage-maven-layout-provider/src/test/java/org/carlspring/strongbox/io/ArtifactOutputStreamTest.java

```diff
@@ -107,26 +74,26 @@ public void testCreateWithTemporaryLocation(TestInfo testInfo)
 
         assertTrue(Files.exists(artifactPathTemp), "Failed to create temporary artifact file!");
 
-
-
         artifactPathTemp.getFileSystem().provider().moveFromTemporaryDirectory(artifactPathTemp);
 
         assertTrue(Files.exists(artifactPath), "Failed to the move temporary artifact file to original location!");
     }
 
     @Test
-    public void testCreateWithTemporaryLocationNoMoveOnClose(TestInfo testInfo)
+    @ExtendWith({ RepositoryManagementTestExecutionListener.class, ArtifactManagementTestExecutionListener.class })
+    public void testCreateWithTemporaryLocationNoMoveOnClose(@TestRepository(repository = "aost=tcwtlnmoc-releases",
+                                                                             layout = Maven2LayoutProvider.ALIAS)
+                                                             Repository repository,
+                                                             @MavenTestArtifact(id = "org.carlspring.foo:temp-file-test",
+                                                                                versions = { "1.2.4" })
+                                                             Path path)
```

Comment:
@sbespalov ,

I'm wondering whether it would be a better idea to make the list of `Path`-s a list of `RepositoryPath`-s`.

Reply:
> I'm wondering whether it would be a better idea to make the list of Path-s a list of RepositoryPath-s`...?

This is not possible at the moment. The thing is that `Path` instance that you have injected here is lazy initialized JDK Proxy (see [here](https://github.com/strongbox/strongbox/blob/master/strongbox-storage/strongbox-storage-api/src/test/java/org/carlspring/strongbox/testing/artifact/ArtifactManagementTestExecutionListener.java#L89)). 
Currently you can get the repository path like this:
```
RepositoryPath repositoryPath = (RepositoryPath)path.normalize();
```

To be able to directly inject `RepositoryPath` we should expose the `RepositoryPath` interface, currently it's just a class.

Reply:
I see! Thanks for the clarification! :)